pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -70,8 +71,10 @@ const (
 	// were to continuously return a nextPageToken.
 	maxPages = 25
 
-	// Target Pool creation is limited to 200 instances.
 	maxTargetPoolCreateInstances = 200
+
+	// Max size of names accepted by the GCE api.
+	nameLengthLimit = 63
```

Comment:
are you doing a corresponding name length cap on the client side? or will it just say "i don't actually have a service with this shortened name, return 404" ?